pkg/scheduler/framework/plugins/serviceaffinity/service_affinity_test.go

```diff
@@ -179,3 +183,269 @@ func TestServiceAffinity(t *testing.T) {
+			services: []*v1.Service{{Spec: v1.ServiceSpec{Selector: map[string]string{"baz": "blah"}}}},
+			expectedList: []framework.NodeScore{{Name: "machine11", Score: 33}, {Name: "machine12", Score: 33},
+				{Name: "machine21", Score: 66}, {Name: "machine22", Score: 66},
+				{Name: "machine01", Score: 0}, {Name: "machine02", Score: 0}},
+			name: "service with partial pod label matches",
+		},
+		{
+			pod: &v1.Pod{ObjectMeta: metav1.ObjectMeta{Labels: labels1}},
+			pods: []*v1.Pod{
+				{Spec: zone0Spec, ObjectMeta: metav1.ObjectMeta{Labels: labels1}},
+				{Spec: zone1Spec, ObjectMeta: metav1.ObjectMeta{Labels: labels1}},
+				{Spec: zone2Spec, ObjectMeta: metav1.ObjectMeta{Labels: labels1}},
+				{Spec: zone2Spec, ObjectMeta: metav1.ObjectMeta{Labels: labels1}},
+			},
+			nodes:    labeledNodes,
+			labels:   []string{"zone"},
+			services: []*v1.Service{{Spec: v1.ServiceSpec{Selector: labels1}}},
+			expectedList: []framework.NodeScore{{Name: "machine11", Score: 75}, {Name: "machine12", Score: 75},
+				{Name: "machine21", Score: 50}, {Name: "machine22", Score: 50},
+				{Name: "machine01", Score: 0}, {Name: "machine02", Score: 0}},
+			name: "service pod on non-zoned node",
+		},
+		{
+			pod: &v1.Pod{ObjectMeta: metav1.ObjectMeta{Labels: labels1}},
+			pods: []*v1.Pod{
+				{Spec: zone0Spec, ObjectMeta: metav1.ObjectMeta{Labels: labels2}},
+				{Spec: zone1Spec, ObjectMeta: metav1.ObjectMeta{Labels: labels1}},
+				{Spec: zone2Spec, ObjectMeta: metav1.ObjectMeta{Labels: labels1}},
+			},
+			nodes:    nodesWithZoneAndRackLabels,
+			labels:   []string{"zone", "rack"},
+			services: []*v1.Service{{Spec: v1.ServiceSpec{Selector: labels1}}},
+			expectedList: []framework.NodeScore{{Name: "machine11", Score: 25}, {Name: "machine12", Score: 75},
+				{Name: "machine21", Score: 25}, {Name: "machine22", Score: 25},
+				{Name: "machine01", Score: 0}, {Name: "machine02", Score: 0}},
+			name: "three pods, two service pods, with rack label",
+		},
+	}
+	// these local variables just make sure controllerLister\replicaSetLister\statefulSetLister not nil
+	// when construct metaDataProducer
+	sss := []*apps.StatefulSet{{Spec: apps.StatefulSetSpec{Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"foo": "bar"}}}}}
+	rcs := []*v1.ReplicationController{{Spec: v1.ReplicationControllerSpec{Selector: map[string]string{"foo": "bar"}}}}
+	rss := []*apps.ReplicaSet{{Spec: apps.ReplicaSetSpec{Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"foo": "bar"}}}}}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			nodes := makeLabeledNodeList(test.nodes)
+			snapshot := nodeinfosnapshot.NewSnapshot(test.pods, nodes)
+			fh, _ := framework.NewFramework(nil, nil, nil, framework.WithSnapshotSharedLister(snapshot))
+			serviceLister := fakelisters.ServiceLister(test.services)
+			priorityMapFunction, priorityReduceFunction := priorities.NewServiceAntiAffinityPriority(snapshot.Pods(), serviceLister, test.labels)
```

Comment:
that is why the tests are passing, we are not using New to instantiate the priority.

Reply:
yeah... I will leave it as is and create a follow up to update the tests - creating a fake SharedInformerFactory is not easy.

Reply:
Created a follow up issue #84943